src/main/c/cext/ruby.c

```diff
@@ -1055,7 +1055,9 @@ VALUE rb_str_new_cstr(const char *string) {
   if (polyglot_is_value((VALUE) string)) {
     VALUE ruby_string = (VALUE) polyglot_invoke((VALUE) string, "to_s");
     int len = strlen(string);
-    return (VALUE) polyglot_invoke(ruby_string, "[]", 0, len);
+    ruby_string = polyglot_invoke(ruby_string, "[]", 0, len);
+    rb_enc_associate(ruby_string, rb_ascii8bit_encoding());
```

Comment:
`String#b` actually creates a new String, is that any better than rb_enc_associate()?
`strlen()` is needed because we need to stop at the first `\0` byte with `rb_str_new_cstr()`.

Reply:
Okay. I didn't realize you needed to deal with strings with larger capacity -- no need to use `b` then. But, shouldn't the `[]` be a `byteslice` call? `strlen` is going to return byte count, not character count.

Reply:
It's switched to rb_str_subseq(), the same as byteslice in 8830af3c1efeb641e71c42183ccdae0ddb61b009